### PR TITLE
Move over to subject categories in curriculum visualiser

### PR DESCRIPTION
**Pull request by orangemug**
## Description
Move over from tags to subject categories in curriculum visualiser

## Issue(s)

Fixes `CUR-786`

## How to test

1. Go to https://deploy-preview-2678--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Test the previous domains/disciplines are still working under new subject categories

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx [hunk 1]
```diff
@@ -294,13 +285,13 @@ const CurriculumVisualiser: FC<CurriculumVisualiserProps> = ({
                 >
                   Year {year}
                 </OakHeading>
-                {disciplines?.length > 1 && (
-                  <Box role="group" aria-label="Disciplines">
-                    {disciplines.map((discipline, index) => {
-                      const isSelected = isSelectedDiscipline(
+                {subjectCategories?.length > 1 && (
+                  <Box role="group" aria-label="Subject category">
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx:289]: For consistency with Pupil squad we should go with an ARIA label of 'Categories' - note also the pluralisation as we're naming the group not the items within it.

File: src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx [hunk 2]
```diff
@@ -71,10 +70,12 @@ type CurriculumVisualiserProps = {
   selectedYear: string | null;
   examboardSlug: string | null;
   yearData: YearData;
-  handleSelectDomain: (year: string, domain: Domain) => void;
   handleSelectSubject: (year: string, subject: Subject) => void;
   handleSelectTier: (year: string, tier: Tier) => void;
-  handleSelectDiscipline: (year: string, discipline: Discipline) => void;
+  handleSelectSubjectCatgeory: (
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx:75]: Typo in name of prop

File: src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx [hunk 3]
```diff
@@ -257,8 +245,11 @@ const CurriculumVisualiser: FC<CurriculumVisualiserProps> = ({
         Object.keys(yearData)
           .filter((year) => !selectedYear || selectedYear === year)
           .map((year, index) => {
-            const { units, childSubjects, domains, tiers, disciplines } =
-              yearData[year] as YearData[string];
+            const { units, childSubjects, tiers, subjectCategories } = yearData[
+              year
+            ] as YearData[string];
+
+            console.log({ year, subjectCategories });
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx:252]: I assume this is going away soon 😄

File: src/node-lib/curriculum-api-2023/queries/curriculumUnits/curriculumUnits.query.ts [hunk 4]
```diff
@@ -51,6 +51,7 @@ const curriculumUnitsQuery =
       throw new OakError({ code: "curriculum-api/not-found" });
     }
 
+    console.log("res=", JSON.stringify(res, null, 2));
```
Comment (rightsaidjames) [inline src/node-lib/curriculum-api-2023/queries/curriculumUnits/curriculumUnits.query.ts:54]: Ditto this console.log

**Review by sonalisin — APPROVED** [review]
Looks good to me, I figure that it's `subjectcategories` on the DB side, but for future we should really update it to `subject_categories` so it's consistent with the other fields

**Pull request merged**